paddle/fluid/framework/operator.cc

```diff
@@ -1274,6 +1274,11 @@ void OperatorWithKernel::ParseInputDataType(
         t = &(var->Get<SelectedRows>().value());
       } else if (var->IsType<LoDTensorArray>()) {
         auto t_arr = var->Get<LoDTensorArray>();
+        // NOTE(liym27): Support an empty tensor array as Input.
+        // And set the kernel type is float.
+        if (t_arr.size() == 0) {
+          *data_type = proto::VarType::FP32;
```

Comment:
Is this necessary? I understand that the program may be wrong if we don't set it, but it is unclear whether set default kernel type to FP32 will cause problem.

I'm fine with the change here but just asked is there other way to do so?

Reply:
> Is this necessary? I understand that the program may be wrong if we don't set it, but it is unclear whether set default kernel type to FP32 will cause problem.
> I'm fine with the change here but just asked is there other way to do so?

Because empty tensor array is used in assign op and assign op has kernel, I will try to  set the kenel type in assign op when tensor array is empty .

Reply:
OK, LGTM